src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -188,6 +188,21 @@ int halide_hexagon_remote_set_performance_mode(int mode) {
     HAP_power_response_t power_info;
     unsigned int max_mips = 0;
     uint64 max_bus_bw = 0;
+    HAP_power_request_t request;
+
+    const HAP_dcvs_voltage_corner_t voltage_corner[]
+        = { HAP_DCVS_VCORNER_SVS,
+            HAP_DCVS_VCORNER_NOM,
+            HAP_DCVS_VCORNER_TURBO,
+            HAP_DCVS_VCORNER_DISABLE,
+            HAP_DCVS_VCORNER_SVSPLUS,
+            HAP_DCVS_VCORNER_SVS2,
+            HAP_DCVS_VCORNER_NOMPLUS };
+
+    if (mode > 6) {
+        log_printf("Unknown power mode (%d)\n");
+        return -1;
+    }
 
     power_info.type = HAP_power_get_max_mips;
     int retval = HAP_power_get(NULL, &power_info);
```

Comment:
It looks like this (and all of it's associated variables) only get used if the DCVS_v2 call fails. I think we should move all the new stuff for DCVS_v2 to happen first, and only do the rest of this if it fails. So I think the structure of this function should be:

    // new code
    if (0 == HAP_power_set(DCVS_v2) {
      return 0;
    }

    // All the old code here that falls through to set_performance for compatibility.

Reply:
Only parameter which needs to be set in both DCVS_V2 and older implementation is latency which is getting determined in the switch case ladder. I can make some changes to determine latency earlier so that DCVS_V2 is done prior to any other stuff. However, i would resist those code movements for the sake of simplicity.